torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1385,8 +1384,10 @@ bool ProcessGroupNCCL::dumpDebuggingInfo() {
     LOG(INFO) << logPrefix() << "ProcessGroupNCCL dumping nccl trace to "
               << writer.getWriterTarget();
     writer.write(ncclTrace);
+    promiseFlightRecorderDump_.set_value(true);
     return true;
   }
+  promiseFlightRecorderDump_.set_value(false);
```

Comment:
actually, is it better we just set_value inside monitoring thread?

Reply:
Good idea! Done!